connaisseur/notary_api.py

```diff
@@ -160,21 +125,23 @@ def parse_auth(auth_header: str):
     return url
 
 
-def get_auth_token(url: str):
+def get_auth_token(notary_config: Notary, url: str):
     """
     Return the JWT from the given `url`, using user and password from
     environment variables.
 
     Raises an exception if a HTTP error status code occurs.
     """
-    user = os.environ.get("NOTARY_USER", False)
-    password = os.environ.get("NOTARY_PASS", "")
     request_kwargs = {"url": url}
-    if user:
-        request_kwargs["auth"] = requests.auth.HTTPBasicAuth(user, password)
 
-    if is_notary_selfsigned():
-        request_kwargs["verify"] = "/etc/certs/notary.crt"
+    if notary_config.has_auth:
+        auth = notary_config.get_auth()
+        request_kwargs["auth"] = requests.auth.HTTPBasicAuth(
+            auth.get("USER"), auth.get("PASS")
+        )
```

Comment:
If you get rid of `has_auth` as suggested above, you may simply rewrite this as:

```python
try:
    request_kwargs["auth"] = requests.auth.HTTPBasicAuth(*notary_config.auth)
except TypeError:  # happens if .auth is None
    pass  # just ignore the failure
```

This is another instance of begging for forgiveness instead of asking for permission.